docs/tasks/administer-cluster/kubeadm-upgrade-ha.md

```diff
@@ -0,0 +1,176 @@
+{: .caution}
+
+## Upgrading your control plane
+
+The following procedure must be applied on a single master node and repeated for each subsequent master node sequentially.
+
+Before initiating the upgrade with `kubeadm` `configmap/kubeadm-config` needs to be modified for the current master host. Replace any hard reference to a master host FQDN with the current master hosts' FQDN:
+
+```shell
+kubectl get configmap -n kube-system kubeadm-config -o yaml >/tmp/kubeadm-config-cm.yaml
+sed -i 's/^\([ \t]*nodeName:\).*/\1 <CURRENT-MASTER-FQDN>/' /tmp/kubeadm-config-cm.yaml
+kubectl apply -f /tmp/kubeadm-config-cm.yaml --force
+```
+
+Now the upgrade process can start. Use the target version determined in the preparation step and run the following command (press “y” when prompted):
+
+```shell
+/tmp/kubeadm upgrade apply v<YOUR-CHOSEN-VERSION-HERE>
+```
+
+If the operation was successful you’ll get a message like this:
+
+    [upgrade/successful] SUCCESS! Your cluster was upgraded to "v1.9.2". Enjoy!
+
+    To upgrade the cluster with CoreDNS as the default internal DNS, invoke `kubeadm upgrade apply` with the `--feature-gates=CoreDNS=true` flag.
+
+`kubeadm upgrade apply` does the following:
+
+- Checks that your cluster is in an upgradeable state:
+  - The API server is reachable,
+  - All nodes are in the `Ready` state
+  - The control plane is healthy
+- Enforces the version skew policies.
+- Makes sure the control plane images are available or available to pull to the machine.
+- Upgrades the control plane components or rollbacks if any of them fails to come up.
+- Applies the new `kube-dns` and `kube-proxy` manifests and enforces that all necessary RBAC rules are created.
+- Creates new certificate and key files of apiserver and backs up old files if they're about to expire in 180 days.
+
+Next, manually upgrade your CNI provider
+
+Your Container Network Interface (CNI) provider may have its own upgrade instructions to follow. Check the [addons](/docs/concepts/cluster-administration/addons/) page to find your CNI provider and see if there are additional upgrade steps necessary.
+
+**Note:** The `kubeadm upgrade apply` step has been known to fail when run initially on the secondary masters (timed out waiting for the restarted static pods to come up). It should succeed if retried after a minute or two.
+{: .note}
+
+## Upgrade base software packages
+
+At this point all the static pod manifests in your cluster, for example API Server, Controller Manager, Scheduler, Kube Proxy have been upgraded, however the base software, for example `kubelet`, `kubectl`, `kubeadm` installed on your nodes’ OS are still of the old version. For upgrading the base software packages on all nodes we will `cordon` them if master and `drain` if worker node, upgrade and then `uncordon` them again one by one. 
```

Comment:
You don't need todo this.  All the packages and daemons allow for them to be in-place upgraded and you don't need to drain.

Reply:
@timothysc to clarify: is it just draining that's unnecessary, or is cordoning also unnecessary? This is great, saves some complexity.

Reply:
Do I understand this correctly: no draining, no cordoning, just upgrade the packages and restart the daemons?